### PR TITLE
AB#85937 AB#85936 Fix for external status and add unit test

### PR DESCRIPTION
**Pull request by dawid-pawulski**

### CODE REVIEW

File: src/HE.Investment.AHP.Domain.Tests/Application/Entities/ApplicationEntityTests/ApplicationEntityHoldTests.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            .Register(this)
+            .UserAccountFromMock;
+
+        var applicationRepository = ApplicationRepositoryTestBuilder
+            .New()
+            .ReturnApplication(applicationId, userAccount, application)
+            .BuildIApplicationHoldMockAndRegister(this);
+
+        // when
+        application.Hold(applicationRepository.Object, holdReason, organisationId, CancellationToken.None);
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.Domain.Tests/Application/Entities/ApplicationEntityTests/ApplicationEntityHoldTests.cs:38]: you're missing await in async entity method and here

File: src/HE.Investment.AHP.Domain.Tests/HE.Investment.AHP.Domain.Tests.csproj [hunk 2]
```diff
@@ -47,6 +47,7 @@
 
   <ItemGroup>
     <ProjectReference Include="..\HE.Investment.AHP.Domain\HE.Investment.AHP.Domain.csproj" />
+    <ProjectReference Include="..\HE.Investments.Account.Domain.Tests\HE.Investments.Account.Domain.Tests.csproj" />
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.Domain.Tests/HE.Investment.AHP.Domain.Tests.csproj:50]: hmm, this reference should not be added. If you're using some code from Account.Tests, then it should be in Common.Tests?

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investment.AHP.Domain.Tests/Application/Entities/ApplicationEntityTests/ApplicationEntityHoldTests.cs (hunk 1 above)
Comment (dawid-pawulski) [reply src/HE.Investment.AHP.Domain.Tests/Application/Entities/ApplicationEntityTests/ApplicationEntityHoldTests.cs]: done

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investment.AHP.Domain.Tests/HE.Investment.AHP.Domain.Tests.csproj (hunk 2 above)
Comment (dawid-pawulski) [reply src/HE.Investment.AHP.Domain.Tests/HE.Investment.AHP.Domain.Tests.csproj]: done

**Pull request merged**